### PR TITLE
Fixed some comparison errors for Jacobian functions

### PR DESCRIPTION
**Pull request by raneamri**
## Type of change

Simple recast on some comparisons that triggered warnings with MSVC2022.

-   Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Review by xcelerit-team — CHANGES_REQUESTED** [review]
Thanks for that. 

Could you use `std::size_t` for the type instead? This is also the type used for the `size_type` of STL containers.

File: src/XAD/Hessian.hpp [hunk 1]
```diff
@@ -54,10 +54,10 @@ void computeHessian(const std::vector<AReal<FReal<T>>> &vec,
                     RowIterator first, RowIterator last,
                     Tape<FReal<T>> *tape = Tape<FReal<T>>::getActive())
 {
-    unsigned int domain(static_cast<unsigned int>(vec.size()));
+    std::size_t domain(static_cast<std::size_t>(vec.size()));
```
Comment (auto-differentiation-dev) [inline src/XAD/Hessian.hpp:57]:
```suggestion
    std::size_t domain = vec.size();
```

File: src/XAD/Hessian.hpp [hunk 2]
```diff
@@ -107,22 +107,22 @@ void computeHessian(const std::vector<FReal<FReal<T>>> &vec,
                     std::function<FReal<FReal<T>>(std::vector<FReal<FReal<T>>> &)> foo,
                     RowIterator first, RowIterator last)
 {
-    unsigned int domain(static_cast<unsigned int>(vec.size()));
+    std::size_t domain(static_cast<std::size_t>(vec.size()));
```
Comment (auto-differentiation-dev) [inline src/XAD/Hessian.hpp:110]:
```suggestion
    std::size_t domain = vec.size();
```

File: src/XAD/Jacobian.hpp [hunk 3]
```diff
@@ -69,19 +69,19 @@ void computeJacobian(const std::vector<AReal<T>> &vec,
     tape->registerInputs(v);
     tape->newRecording();
     auto y = foo(v);
-    unsigned int domain = static_cast<unsigned int>(vec.size()),
-                 codomain = static_cast<unsigned int>(y.size());
-    if (std::distance(first, last) != codomain)
+    std::size_t domain = static_cast<std::size_t>(vec.size()),
+                 codomain = static_cast<std::size_t>(y.size());
```
Comment (auto-differentiation-dev) [inline src/XAD/Jacobian.hpp:73]:
```suggestion
    std::size_t domain = vec.size();
    std::size_t codomain = y.size();
```

**Review by auto-differentiation-dev — CHANGES_REQUESTED** [review]
Just a few simplifications, then it's good to go

**Review by xcelerit-team — APPROVED** [review]

**Pull request merged**